price-feeder/oracle/oracle.go

```diff
@@ -356,6 +370,54 @@ func SetProviderTickerPricesAndCandles(
 	return pricesOk || candlesOk
 }
 
+// Update update the values inside the OnChainData struct.
+func (onChainData *OnChainData) Update(currentBlockHeigh uint64, params oracletypes.Params) {
+	onChainData.lastUpdatedBlock = currentBlockHeigh
+	onChainData.params = &params
+}
+
+func (onChainData *OnChainData) IsParamsOutdated(currentBlockHeigh uint64) bool {
+	// doesn't have any data
+	if onChainData.params == nil {
+		return true
+	}
+
+	// state in which the blockchain is under cacheOnChainBlockQuantity
+	// amount of blocks, but already has value for params
+	if currentBlockHeigh < cacheOnChainBlockQuantity {
+		return false
+	}
+
+	// an edge case, should never happen
+	// which the current blockchain height is lower
+	// than the last updated block, to fix should
+	// just update the on chain params again
```

Comment:
When does this happen? Are you just making sure we're covering all code paths? 

```suggestion
	// This is an edge case, which should never happen.
	// The current blockchain height is lower
	// than the last updated block, to fix we should
	// just update the cached params again.
```